packages/next/client/image.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+
+const loaders: { [key: string]: (props: LoaderProps) => string } = {
+  imgix: imgixLoader,
+  cloudinary: cloudinaryLoader,
+  default: defaultLoader,
+}
+type ImageData = {
+  hosts: {
+    [key: string]: {
+      path: string
+      loader: string
+    }
+  }
+  breakpoints?: number[]
+}
+type ImageProps = {
+  src: string
+  host: string
+  sizes: string
+  breakpoints: number[]
+  unoptimized: boolean
+  rest: any[]
+}
+
+let imageData: any = process.env.__NEXT_IMAGE_OPTS
+const breakpoints = imageData.breakpoints || [640, 1024, 1600]
+
+function computeSrc(src: string, host: string, unoptimized: boolean): string {
+  if (unoptimized) {
+    return src
+  }
+  if (!host) {
+    // No host provided, use default
+    return callLoader(src, 'default')
+  } else {
+    let selectedHost = imageData.hosts[host]
+    if (!selectedHost) {
+      console.error(
```

Comment:
This needs to be protected by some sort of isDev flag. We would not want these strings in client bundle on production.

Reply:
`if(typeof process.env.NODE_ENV !== 'production)` would be the right way